cmd/mattermost/commands/command.go

```diff
@@ -67,3 +89,80 @@ func moveCommandCmdF(command *cobra.Command, args []string) error {
 func moveCommand(a *app.App, team *model.Team, command *model.Command) *model.AppError {
 	return a.MoveCommand(team, command)
 }
+
+func createCommandCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("enter the team this command will be created for")
+	}
+
+	team := getTeamFromTeamArg(a, args[0])
+	if team == nil {
+		return errors.New("unable to find team '" + args[0] + "'")
+	}
+
+	title, _ := command.Flags().GetString("title")
+	description, _ := command.Flags().GetString("description")
+	trigger, errt := command.Flags().GetString("trigger-word")
+	if errt != nil || trigger == "" {
+		return errors.New("atrigger word is required")
+	}
+	if strings.HasPrefix(trigger, "/") {
+		return errors.New("a trigger word cannot begin with a /")
+	}
+	if strings.Contains(trigger, " ") {
+		return errors.New("a trigger word must not contain spaces")
+	}
+
+	url, erru := command.Flags().GetString("url")
+	if erru != nil || url == "" {
+		return errors.New("a request URL is required")
+	}
+	creator, errc := command.Flags().GetString("creator")
+	if errc != nil || creator == "" {
+		return errors.New("a creator username is required")
+	}
+	user := getUserFromUserArg(a, creator)
+	responseUsername, _ := command.Flags().GetString("response-username")
+	icon, _ := command.Flags().GetString("icon")
+	autocomplete, _ := command.Flags().GetBool("autocomplete")
+	autocompleteDesc, _ := command.Flags().GetString("autocompleteDesc")
+	autocompleteHint, _ := command.Flags().GetString("autocompleteHint")
+	post, errp := command.Flags().GetBool("post")
+	method := "P"
+	if errp != nil || post == false {
+		method = "G"
+	}
+
+	newCommand := &model.Command{
+		CreatorId:        user.Id,
+		TeamId:           team.Id,
+		Trigger:          trigger,
+		Method:           method,
+		Username:         responseUsername,
+		IconURL:          icon,
+		AutoComplete:     autocomplete,
+		AutoCompleteDesc: autocompleteDesc,
+		AutoCompleteHint: autocompleteHint,
+		DisplayName:      title,
+		Description:      description,
+		URL:              url,
+	}
+
+	if err := createCommand(a, team, newCommand); err != nil {
+		return errors.New("unable to create command '" + newCommand.Trigger + "'. " + err.Error())
+	}
+	CommandPrettyPrintln("created command '" + newCommand.Trigger + "'")
+
+	return nil
+}
+
+func createCommand(a *app.App, team *model.Team, command *model.Command) *model.AppError {
```

Comment:
I don't get the point of this wrapper, we can use directly the `a.CreateCommand`, is there any reason to put it in a separate function?

Reply:
I wasn't exactly sure either, but I was trying keep conformity with the way the `move` command was setup.  I can certainly remove it.

Reply:
Yes, I think in this cases doesn't provide any value and can generate some confusion.